docs/content/rust/guides/roles/getting-started.md

```diff
@@ -0,0 +1,274 @@
+`roles` are unbound parameters, meaning they can be assigned inside of
+the rule body.
+
+We could have written this rule without a body:
+
+```polar
+resource( _type: Org, "org", ["read", "create_repo"],
+    {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["read", "create_repo"],
+        }
+    }
+);
+```
+
+But we think the expanded form is clearer.
+
+{{% /callout %}}
+
+### Adding `role_allows` to our policy
+
+To allow access based on roles, we add the following `allow` rule
+
+```polar
+allow(actor, action, resource) if
+    # add other conditions can be added here if needed
+    role_allows(actor, action, resource);
+```
+
+Oso will now allow access to any resource that is allowed based on the
+role definitions. 
+
+### Assigning roles to users
+
+Now we've configured roles and set up our policy. For users to have
+access, we must assign them roles.
+
+You can use your own data models for roles with Oso. You just need to tell us
+what roles a user has for a particular resource through the
+`actor_has_role_for_resource` rule. As an example, our types might have an
+interface like this:
+
+```rust
+pub struct OrgRole {
+  pub name: String,
+  pub resource: Org,
+}
+
+impl User {
+  pub fn roles(&self) -> Vec<OrgRole> {
+    // ...
+  }
+}
+```
+
+And the `actor_has_role_for_resource` would be implemented as:
+
+```polar
+actor_has_role_for_resource(actor, role_name, resource) if
+    role in actor.roles and
+    role_name = role.name and
+    resource = role.resource;
+```
+
+The `actor_has_role_for_resource` is evaluated with `actor` bound to the same actor
+that you call the `allow` rule with, typically an instance of some `User` model.
+
+`role_name` and `resource` are "output parameters".
+In the body of the `actor_has_role_for_resource` rule, you
+should unify `role_name` with the name of the actor's role and
+`resource` with the instance the actor has the role for.
+
+### Implying roles
+
+The `"owner"` role is a more permissive role than `"member"`. It
+covers all the permissions of `"member"`, with some additional
+permissions granted (`"create_repo"`) in our example.
+
+Instead of duplicating the permissions, we can represent this
+relationship in our policy using **implied roles**.
+
+```polar
+resource(_type: Org, "org", actions, roles) if
+    actions = ["read", "create_repo"] and
+    roles = {
+        member: {
+            permissions: ["read"],
+        },
+        owner: {
+            permissions: ["create_repo"],
+            implies: ["member"]
+        }
+    };
+```
+
+The `"owner"` role now implies the `"member"` role. Any user with the
+`"owner"` role will be granted all permissions associated with both
+roles.
+
+Here's the full `Org` resource definition from the GitClub example app:
+
+{{< literalInclude
+    path="examples/gitclub-sqlalchemy-flask-react/backend/app/authorization.polar"
```

Comment:
Hmm (not related to this PR) I wonder if these `literalIncludes` will break with the repo name change to `gitclub`. I'll file a pull request to fix all those.

Reply:
👍